crates/core/src/marzano_code_snippet.rs

```diff
@@ -64,11 +64,19 @@ impl Matcher<MarzanoQueryContext> for MarzanoCodeSnippet {
             return Ok(false);
         };
 
-        if let Some((_, pattern)) = self
-            .patterns
-            .iter()
-            .find(|(id, _)| *id == node.node.kind_id())
-        {
+        if let Some((_, pattern)) = self.patterns.iter().find(|(id, p)| {
+            let kind_id = node.node.kind_id();
+            if *id == kind_id {
+                return true;
+            }
+            // use equivalence classes to match 'ubuntu-latest' and "ubuntu-latest" in yaml
+            // i.e. to match string_scalar, single_quote_scalar, and double_quote_scalar
+            // see https://github.com/getgrit/gritql/issues/394
+            match p {
+                Pattern::AstLeafNode(p) => p.is_equivalent_class(kind_id),
+                _ => false, // TODO: handle other pattern types?
```

Comment:
Our style guide requires expanding this so each arm is explicit.

Reply:
ok, no worries have done: bf9bb755

I don't yet know in depth, what all the different Pattern enum variants mean. From my understanding, it seemed like it was less relevant for the others so I have just returned `false` for all of the other variants - let me know if that doesn't make sense

Reply:
Yeah that's fine for now. We just like to enumerate them so explicit choices are made when adding new ones.

Reply:
`AstLeafNode` seems to be the only enum variant that has a concept of
`equivalence_class: Option<LeafEquivalenceClass>`

Reply:
ok, sounds good!